### PR TITLE
Fix a couple issues with npm/apm parsing

### PR DESCRIPTION
**Pull request by tresni**
FAO: @kdeldycke

A couple tweaks after investigating what @keithamus brought up with linked packages.

### CODE REVIEW

File: Dev/Homebrew/package_manager.7h.py [hunk 1]
```diff
@@ -214,7 +214,12 @@ def name(self):
     def sync(self):
         output = self.run(self.cli, '-g', '--progress=false', '--json',
                           'outdated')
+        if not output:
+            return
+
         for package, values in json.loads(output).iteritems():
+            if values['current'] == 'link':
```
Comment (keithamus) [inline Dev/Homebrew/package_manager.7h.py]: needs to be `values['wanted'] == 'linked'`. I know this seems very counter intuitive, but the `current` is the version that is in the package.json, which is likely never `linked`.

**Comment by tresni** [comment]
gotta update for @kdeldycke's update anyway so I'll get his (and a `sudo` on the gem commands)

**Comment by tresni** [comment]
rebased on latest master and using wanted instead of installed.

**Comment by tresni** [comment]
Added an update for Gems support when using system ruby.

File: Dev/Homebrew/package_manager.7h.py [hunk 2]
```diff
@@ -371,7 +371,12 @@ def sync(self):
         """
         output = self.run(
             self.cli, '-g', '--progress=false', '--json', 'outdated')
+        if not output:
+            return
+
         for package, values in json.loads(output).iteritems():
+            if values['wanted'] == 'link':
```
Comment (keithamus) [inline Dev/Homebrew/package_manager.7h.py]: `'linked'` not `'link'` btw.

**Comment by kdeldycke** [comment]
Thanks @tresni for the follow-up! :)

**Pull request merged**